### PR TITLE
Add README examples for blocks, methods, and options hash

### PR DESCRIPTION
**Pull request by ghost**
Here's more for your users... hope this is helpful!  Thanks, Joel

### CODE REVIEW

File: README.rdoc [hunk 1]
```diff
@@ -56,6 +56,52 @@ Shorthand for create:
...
+
+This initializes the object, then calls the "name=" method on the object.
+
+
+== Blocks + Methods 
+
+To use a block and method together: 
+
+    Factory.define :user do |f|
+      f.name {|x| x.name || "Alice" }
```
Comment (stephencelis) [inline README.rdoc]: While it's interesting that this example works, I'm not sure it should be documented over the simpler:

``` ruby
Factory.define :user do |f|
  f.name 'Alice'
end

u = Factory :user
u.name # => "Alice"

u = Factory :user, name: 'Bob'
u.name # => "Bob"
```

Comment (stephencelis) [inline README.rdoc]: (Maybe it makes more sense to provide an example that requires a block? Like a timestamp?)

File: README.rdoc [hunk 2]
```diff
@@ -56,6 +56,52 @@ Shorthand for create:
...
+== Methods
+
+To call a method on an object as it's being built:
+
+    Factory.define :user do |f|
+    end
+
+    Factory(:user, :name => "Bob")
+
+This initializes the object, then calls the "name=" method on the object.
```
Comment (stephencelis) [inline README.rdoc]: This is good to point out (and is the same behavior as Factory Girl, I believe), but again, I'm not sure about the example. Something like `name=` seems like a simple `attr_accessor`...maybe an override that does more than (what I assume) merely assigns an ivar?

**Comment by stephencelis** [comment]
I appreciate the initiative in documentation, thanks! :) I have a few notes about the examples, though. Let me know what you think!

File: README.rdoc (hunk 1 above)
Comment (SixArm) [inline README.rdoc]: Good point. Yes, I'll change it to a timestamp.

File: README.rdoc (hunk 2 above)
Comment (SixArm) [inline README.rdoc]: Agreed. I can create a better example and will push it -- thanks for reviewing this.

**Comment by SixArm** [comment]
Thanks Stephen, here are some better examples that are closer to my real-world use.  -Joel

**Comment by stephencelis** [comment]
Thanks!

**Pull request merged**